app/src/main/assets/viewer.js

```diff
@@ -198,15 +198,28 @@ function isTextSelected() {
     return window.getSelection().toString() !== "";
 }
 
-pdfjsLib.getDocument("https://localhost/placeholder.pdf").promise.then(function(newDoc) {
-    pdfDoc = newDoc;
-    channel.setNumPages(pdfDoc.numPages);
-    pdfDoc.getMetadata().then(function(data) {
-        channel.setDocumentProperties(JSON.stringify(data.info));
-    }).catch(function(error) {
-        console.log("getMetadata error: " + error);
+function loadDocument() {
+    const pdfPassword = channel.getPassword();
+    let loadingTask = pdfjsLib.getDocument({ url: "https://localhost/placeholder.pdf", password: pdfPassword });
+    loadingTask.onPassword = (_, error) => {
+        if (error == pdfjsLib.PasswordResponses.NEED_PASSWORD) {
```

Comment:
Try to use triple equals when possible as it checks for equality and data type.

e.g.

```
const test1 = 1 // int
const test2 = "1" // string

test1 == test2 // true
test1 === test2 // false
```